src/Compilers/Core/Portable/RuntimeCapability.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies capabilities that may or may not be supported by the common language runtime the compilation is
+    /// targeting.
+    /// </summary>
+    public enum RuntimeCapability
+    {
+        /// <summary>
+        /// Represents a runtime feature where types can define ref fields.
+        /// </summary>
+        ByRefFields = 1,
+
+        /// <summary>
+        /// Represents a runtime feature where overriding methods can return more derived types than the method they override.
+        /// </summary>
+        CovariantReturnsOfClasses = 2,
+
+        /// <summary>
+        /// Represents a runtime feature where interfaces can define default method implementations.
+        /// </summary>
+        DefaultImplementationsOfInterfaces = 3,
+
+        /// <summary>
+        /// Indicates that this version of the runtime supports IntPtr and UIntPtr as numeric types.
+        /// </summary>
+        NumericIntPtr = 4,
+
+        /// <summary>
+        /// Represents a runtime feature where C# function pointers can be declared with an unmanaged calling convention.
+        /// </summary>
+        UnmanagedSignatureCallingConvention = 5,
+
+        /// <summary>
+        /// Indicates that this version of runtime supports virtual static members of interfaces.
+        /// </summary>
+        VirtualStaticsInInterfaces = 6,
+    }
+
+    /// <summary>
+    /// Shared helpers to determine if a runtime capability is supported.  Located in a shared location so that we
+    /// return the same result per runtime, regardless of whether this is a C# or VB compilation.
+    /// </summary>
+    internal static class RuntimeCapabilityHelpers
```

Comment:
>RuntimeCapabilityHelpers [](http://example.com/codeflow?start=26&length=24)

I do not like this refactoring. It creates to much unnecessary churn. My reference would be to keep each language doing its own work and not adding more APIs to ```IAssemblySymbolInternal```. When looking at the change, I would prefer it to be obvious (because there are no code changes) that internal behavior of the compiler is not changed. We are only adding a public API that that calls into existing internal implementation. Obviously, for VB we will need to add more internal implementation.

Reply:
I get that. Though my hope would be that our extensive tests validate that the behavior of the compiler is unchanged. 

The virtue of the refactoring here is that we truly are sharing the same logic between c# and vb. So there's no risk of getting it wrong between them. For example, if one of the checks needs to validate a particular attribute is present, having the logic be unified means we do that properly for both languages (as opposed to potentially forgetting it for one of them). 

Also, since we did say we wanted to be accurate here, regardless of the language, it really does help (and bake a lot of sense) to unify the impl to ensure that language cannot affect this. 

I can change this. But I think it will make it much easier to introduce mistakes in the future.

Reply:
I understand the motivation, but my opinion remains the same. Among other things, I especially dislike extending the ```IAssemblySymbolInternal``` in this fasion.

> Though my hope would be that our extensive tests validate that the behavior of the compiler is unchanged.

No change is the best guarantee for no behavior change.

Reply:
Ok. Just to confirm. You're stating this as more than a preference, and as an explicit change request for the pr?  

If it's just a preference, I'd prefer to keep as is. If it's an explicit request to change it, I'll make the change. 

Let me know which case this is and I'll follow-up accordingly. Thanks!

Reply:
> Just to confirm. You're stating this as more than a preference, and as an explicit change request for the pr?

I will not sign-off on the change as is.